pkg/aggregator/no_aggregation_stream_worker.go

```diff
@@ -152,19 +176,23 @@ func (w *noAggregationStreamWorker) run() {
 							serie.Points = []metrics.Point{{Ts: sample.Timestamp, Value: sample.Value}}
 							serie.Tags = tagset.CompositeTagsFromSlice(w.metricBuffer.Copy())
 							serie.Host = sample.Host
-							serie.MType = sample.Mtype.ToAPIType()
+							serie.MType = mtype
 							// ignored by the intake when late but mimic dogstatsd traffic here anyway
 							serie.Interval = 10
 							w.seriesSink.Append(&serie)
 
 							w.taggerBuffer.Reset()
 							w.metricBuffer.Reset()
+							processed++
 						}
 
 						lastStream = time.Now()
 
-						serializedSamples += len(samples)
+						serializedSamples += processed
+						tlmNoAggSamplesProcessed.Add(float64(processed), noAggProcessStateOk)
```

Comment:
Not sure how frequently this will end up being called, but it still does a synchronized map lookup under the hood which is easily avoidable. Would it be possible to use `Counter#WithTags` to get a counter instance with pre-populated tags at the beginning of the function / in a global variable? ([example](https://github.com/DataDog/datadog-agent/pull/8756/files#diff-d16366ed2d59b497efea1f3bde3296e2944634354e4a4d6adf96078256740393R50-R51))